pkg/conversion/mapi2capi/aws_fuzz_test.go

```diff
@@ -0,0 +1,215 @@
+/*
+Copyright 2024 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package mapi2capi_test
+
+import (
+	"strings"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	fuzz "github.com/google/gofuzz"
```

Comment:
This repo was archived on 10 Jan, 2024. They don't list a reason why, but my assumption is because Go 1.18 included fuzzing.

Did you evaluate Go's built in fuzzing? Or does apimachinery/pkg/api/apitesting/fuzzing use gofuzz?

Reply:
I'm leveraging the archived library as it's used in the Kube and CAPI fuzzing. I expect at some point they will move away from it, but until they do I'm happy to keep this using the same library to be consistent.

Looking at the built in fuzzing, it looks nowhere near as advanced and doesn't look like we can even add custom fuzzer functions, so at the moment, what we have seems like the best choice.

I cannot seem to find any discussion on the K8s issues related to the fuzz library being archived, I'll poke around a bit more

Reply:
I did a quick comparison, I agree that the built in fuzz functionality isn't where we'd like it to be. Moving off the archived repo will likely require someone taking initiative upstream to bring it in tree or create a new repo.